### DISCUSSION TITLE
Pybricks terminal

### DISCUSSION THREAD

**Discussion by chinmaym505** [pybricks/support#1212] [Q&A] [answered]
Could someone kindly provide me with some information regarding the interactivity of the Pybricks terminal? I would like to know if it is possible for a user to interact with the terminal in any way or if it is primarily used for displaying information. Thank you for your help.

**Comment by dlech** [comment]
For hubs other than the move hub, you can click the REPL button in Pybricks Code (looks like `>>>`) and you will get an interactive Python prompt with most things already imported. You you can just type `hub.light.on(Color.GREEN)` and press enter and see what happens.

It is also possible to read from the terminal while a program is running by reading from stdin. You can learn more about this at https://pybricks.com/projects/tutorials/wireless/hub-to-device/

**Comment by chinmaym505** [comment]
What about the terminal commands?

**Reply by dlech** [reply, marked as answer]
If you are using EV3, then it is running Linux and you can use SSH to get a bash terminal.

The newer hubs are much more limited and don't have an OS. So the Python prompt is all you get.

**Comment by chinmaym505** [comment]
oh

**Comment by FLL-Team-24277** [comment]
@dlech , can you explain a little more about the spike hubs "not having an OS"? It seems to me that the firmware is the OS?? That's whether it is the stock Lego firmware or pybricks? Or maybe it isn't really appropriate to think of them as an OS.

Anyway, it sounds like you are saying that we Spike users can click the >>> button in the pybricks web app. Is there a way to get a REPL with a windows powerShell terminal? Or within VS Code?

**Reply by dlech** [reply]
It would probably be more accurate to say that it doesn't have a general-purpose OS (like Linux or Nuttx or even FreeRTOS). But we do have a layer running underneath MicroPython that you could call the operating system.

> Is there a way to get a REPL with a windows powerShell terminal? Or within VS Code?

Not an easy way currently. This has been on my todo list. It's not easy to get asyncio (which we need for Bluetooth comms) to play nice with terminal stdio in Python.

**Comment by chinmaym505** [comment]
Could you try adding firmware to make it Linux

**Reply by dlech** [reply]
There is not nearly enough RAM to run Linux.

**Comment by chinmaym505** [comment]
Oh ok
